python/lsst/ts/wep/donutSizeCorrelator.py

```diff
@@ -0,0 +1,344 @@
+import lsst.afw.image
+import numpy as np
+from lsst.ts.wep.utils import binArray
+from scipy.ndimage import binary_dilation
+from scipy.signal import correlate, find_peaks, peak_prominences
+
+
+class DonutSizeCorrelator:
+    """Methods for estimating donut diameters directly from post_isr_image.
+
+    Spiritually, this algorithm works by finding the brightest donut in the
+    image and fitting an annular aperture to maximize the average flux per
+    pixel.
+
+    The implementation is quite different though, to speed computation.
+    It works by correlating annular templates of a bunch of different sizes,
+    and recording the max correlation in the image. It then looks for a local
+    maximum in this signal. The algorithm uses a dynamically scaling resolution
+    to increase efficiency.
+
+    Example usage (assuming exposure is a post_isr_image object):
+        correlator = DonutSizeCorrelator()
+        image = correlator.prepButlerExposure(exposure)
+        diam, diam_grid, corr_grid = correlator.getDonutDiameter(image)
+        plt.plot(diam_grid, corr_grid)
+        plt.axvline(diam, c="r")
+    """
+
+    @staticmethod
+    def prepButlerExposure(
+        exposure: lsst.afw.image.Exposure,
+        badPixelTypes: tuple[str, str, str, str] = ("SAT", "BAD", "NO_DATA", "INTRP"),
+        nDilation: int = 100,
+        fillVal: float = -10.0,
+    ) -> np.ndarray:
+        """Prep image from the butler for correlation.
+
+        Parameters
+        ----------
+        exposure : lsst.afw.image.Exposure
+            Exposure from the butler.
+        badPixelTypes : tuple[str], optional
+            Names of flags that will be masked.
+            Default is ("SAT", "BAD", "NO_DATA", "INTRP")
+        nDilation : int, optional
```

Comment:
Why such a large number of dilation iterations for a bad pixel?

Reply:
Operating on an entire image this removes a region surrounding it, it doesn't actually end up being a large footprint